mozilla/kitsune
# [bug 785856] Upgrade Celery from v2.5.x to v3.0.x

- Updates billiard and moves it from vendor/packages to vendor/src
- Updates celery to v3.0.24
- Updates django-celery to v3.0.23 (there is no v3.0.24 for django-celery)
- Adds amqp v1.4.2 replacing amqplib v1.0.2
- Small code changes to match the new API for celery
- Migration that adds missing tables for celery

I think that covers it.

You need to run `schematic migrations/` because we were missing 2 tables and south was puking on the migrate. Then you can run south `./manage migrate`.

To verify:
- Run the test suite
- Change `CELERY_ALWAYS_EAGER = False` in settings_local.py and fire up the dev server and celeryd (`./manage.py celeryd`)
- Do things on the site that fire off tasks: answer questions, reply to forum posts, trigger a partial reindex from the admin, etc. Make sure the tasks run by looking at the console where celeryd is running.

I think that covers it. We'll want to triple verify things on stage once it gets there.

r?
